SRC/PORT.CPP

```diff
@@ -224,7 +231,7 @@ void deinit()
     // Uninitialize SDL controllers before main SDL
     delete controllers;
 
-    if ( surface ) SDL_FreeSurface( surface );
+    if ( surface ) SDL_UnlockSurface(surface); SDL_FreeSurface( surface );
```

Comment:
Bug here: FreeSurface is not governed by the if.
```suggestion
    if ( surface ) {
        SDL_UnlockSurface( surface ); 
        SDL_FreeSurface( surface );
    }
```

Reply:
> 
> 
> Bug here: FreeSurface is not governed by the if.

Indeed, good catch! I decided to skip the surface fix from this PR since it's not needed in the end and my fix was only partial. To be fixed properly later in separate PR. Relevant changes reverted.